### PR TITLE
Generator and documentation updates for Hooks

### PR DESCRIPTION
**Pull request by julienben**
## React Boilerplate

**1/** The main change in this branch is that the generators no longer offer you the choice of `Stateless Function` vs `React.PureComponent` vs `React.Component`.

Everything is now a functional component but I added the option to wrap the generated component inside `React.memo`.

I tested but please review this carefully nonetheless.

(Side question: Aren't `PureComponent` and `React.memo` redundant with `reselect`? @jwinn)

**2/** Containers generated with reducers and sagas use the `useInjectReducer` and `useInjectSaga` hooks instead of the HOCs we've had until now.

Note that the HOCs are still there so we still support classes, just like React.

**UPDATE:** I've added tests for the hooks.

**3/** Updates to the generate-templates-for-linting script and the documentation to reflect the aforementioned changes.

- [x] You have followed our [**contributing guidelines**](https://github.com/react-boilerplate/react-boilerplate/blob/master/CONTRIBUTING.md)
- [x] Double-check your branch is based on `dev` and targets `dev` 
- [x] Pull request has tests (we are going for 100% coverage!)
- [x] Code is well-commented, linted and follows project conventions
- [x] Documentation is updated (if necessary)
- [x] Internal code generators and templates are updated (if necessary)
- [x] Description explains the issue/use-case resolved and auto-closes related issues

### CODE REVIEW

File: app/utils/tests/injectReducer.test.js [hunk 1]
```diff
@@ -66,3 +67,32 @@ describe('injectReducer decorator', () => {
     expect(children.props).toEqual(props);
   });
 });
+
+describe('useInjectReducer hook', () => {
```
Comment (gretzky) [inline app/utils/tests/injectReducer.test.js]: i LOVE you for this test

File: internals/generators/component/index.js [hunk 2]
```diff
@@ -37,6 +26,12 @@ module.exports = {
         return 'The name is required';
       },
     },
+    {
```
Comment (gretzky) [inline internals/generators/component/index.js]: i LOVE you for this, part II

File: app/utils/tests/injectReducer.test.js (hunk 1 above)
Comment (julienben) [reply app/utils/tests/injectReducer.test.js]: Made possible by @Mensae's work on migrating to react-testing-library

Comment (gretzky) [reply app/utils/tests/injectReducer.test.js]: wtb @Mensae !!!

**Pull request merged**